File: Intropoo/Ejercicio1/Persona.py
class Persona:
    def __init__(self, edad, nombre, ciudad):
        self.__edad = edad
        self.__nombre = nombre
        self.__ciudad = ciudad

    def mostrar(self):
        return f"Hola, soy {self.__nombre} de {self.__ciudad}"

    def VerificarEdad(self):
        if(self.__edad >= 18):
            print(f"{self.__nombre} es mayor de edad")
        else:
            print(f"{self.__nombre} es menor de edad")
# a) Mostrar saludo
p1 = Persona(21,"carlos","La Paz")
print(p1.mostrar())
# b) Crea 3 persona y su saludo 
p2 = Persona(10,"Mauricio","Beni")
p3 = Persona(18,"Manuel","Santa Cruz")
p4 = Persona(10,"Erland","La Paz")
print(p2.mostrar())
print(p3.mostrar())
print(p4.mostrar())

# c) Verificar si es mayor de edad
p1.VerificarEdad()
p2.VerificarEdad()
p3.VerificarEdad()
p4.VerificarEdad()

File: Intropoo/Ejercicio1/Persona.java
package Intropoo.Ejercicio1;

public class Persona {
    private int edad;
    private String nombre;
    private String ciudad;

    public Persona(int edad, String nombre, String ciudad) {
        this.edad = edad;
        this.nombre = nombre;
        this.ciudad = ciudad;
    }

    public void mostrar() {
       System.out.println("Hola, soy " + nombre + " de " + ciudad);
    }
    
    // c) Verificar si es mayor de edad 
    public void VerificarEdad() {
        if(edad >= 18){
            System.out.println(nombre + " es mayor de edad.");      
        }else {
            System.out.println(nombre + " es menor de edad.");
        }
    }
    


public static void main(String[] args) {
    
            // a) Mostrar el saludo 
    Persona p1 = new Persona(21,"carlos","La Paz");
    p1.mostrar();
    
    // b) Crea 3 personas y crea su saludo 
    Persona p2 = new Persona(10,"Mauricio","Beni");
    Persona p3 = new Persona(18,"Manuel","Santa Cruz");
    Persona p4 = new Persona(10,"Erland","La Paz");
    p2.mostrar();
    p3.mostrar();
    p4.mostrar();
    System.out.println(" ----- EDADES DE LAS PERSONAS -----");
    // c) verificar su edad 
    p1.VerificarEdad();
    p2.VerificarEdad();
    p3.VerificarEdad();
    p4.VerificarEdad();

}
    
}

